src/node-capnp/capnp.js

```diff
@@ -28,6 +28,7 @@ if (typeof CAPNP_NO_DYNAMIC_REQUIRE !== "undefined" && CAPNP_NO_DYNAMIC_REQUIRE)
 } else {
   // Look for binary for this platform
   let targetPlatform = process.env.npm_config_platform || process.platform;
+  let targetArch = process.env.npm_config_arch;
```

Comment:
This variable doesn't seem to get used?

Reply:
It gets used in build.js as args.targetArch

Reply:
Sorry, I'm not sure I follow - this file doesn't drive `build.js` as far as I understand - this file runs at runtime, not at build-time? `targetArch` also doesn't seem to get used anywhere in this file - there's a reference to `process.arch` just below on line 33, but that should be correct at runtime on the target arch?

Reply:
Interesting - the reason it was added is because I needed targetArch in build.js, and the only way I could see to do this was by including in capnp.js. How else is it defined for build.js? Sorry for my lack of clarity here - very much learning!

Reply:
I was trying to follow back how targetPlatform was defined to help provide some context

Reply:
> needed targetArch in build.js

As far as I can see you need `args.targetArch`, which is parsed from the command line option `--target-arch` passed to build.js by yargs on lines 34..49. This command line flag has its value set to:

```
    default: process.env.npm_config_arch || process.arch
```

which will correctly pick up the desired target from the `npm_config_arch` environment variable.

> only way I could see to do this

I guess what I'm saying is that I don't think this is what's doing it

> follow back how targetPlatform was defined

Indeed - but if you look below on line 33, you can see where `targetPlatform` is being used (to determine the path to find the compiled node module at), whereas `targetArch` doesn't get used similarly - it looks up the currently executing architecture at runtime (from `process.arch`). Using `process.arch` isn't a problem (it should always be correct at runtime), but that's the difference between what `targetPlatform` does now

Reply:
Cool - thanks! Done